### PR TITLE
feat(executor): add classifier as new family of executor

### PR DESCRIPTION
**Pull request by hanxiao**
This PR introduces a new family of executor: `Classifier`, which can be used to generated ML/DL-based tags. This is considered as document enrichment; it can enable future features such as smart sub-flow routing.

## Scope

- Add `BaseClassifier` and corresponding drivers to support: binary, multi-class, multi-label, regression (considered as soft label) prediction
- Add testing

## Out of Scope
- Add any concrete classification algorithm

### CODE REVIEW

File: jina/drivers/predict.py [hunk 1]
```diff
@@ -0,0 +1,163 @@
...
+        :param output_tag: output label will be written to ``doc.tags``
+        :param args:
+        :param kwargs:
+        """
+        super().__init__(*args, **kwargs)
+        self.output_tag = output_tag
+
+    def _apply_all(
+            self,
+            docs: Iterable['jina_pb2.Document'],
```
Comment (bwanglzu) [inline jina/drivers/predict.py:34]: i think `Sequence` is a better type hints since we expect the interable object to be ordered

**Review by bwanglzu — COMMENTED** [review]
some other thoughts, maybe we can expose `self.labels` as a property,

File: jina/drivers/predict.py (hunk 1 above)
Comment (hanxiao) [reply jina/drivers/predict.py:34]: Then this should be changed globally in a separate PR

Comment (hanxiao) [reply jina/drivers/predict.py:34]: raised #1196

**Pull request merged**